src/Helpers/RecognizeClassHelper.php

```diff
@@ -14,22 +14,26 @@ public static function recognizeObjectClass(string $objectName): string
             return $objectName;
         }
 
-        $objectName = Str::ucfirst(Str::singular($objectName));
+        $objectName = Str::ucfirst($objectName);
 
-        return self::getObjectNamespace($objectName) . $objectName;
+        return self::getObjectNamespace(Str::singular($objectName)) . $objectName;
     }
 
     public static function getObjectNamespace(string $objectName): string
     {
         $type = self::guessType($objectName);
-        $typeNamespace = self::getNamespaceFromConfig($objectName, $type);
+        $typeNamespaces = config("blt.namespaces.types");
+
+        if (array_key_exists($objectName, $typeNamespaces)) {
+            return $typeNamespaces[$objectName];
+        }
 
-        if ($typeNamespace) {
-            return $typeNamespace;
+        if (array_key_exists($type, $typeNamespaces)) {
+            return $typeNamespaces[$type];
         }
```

Comment:
Maybe getting object from config should be moved to `recognizeObjectClass` instead of this namespace guesser? It would be quicker to go through.